### PR TITLE
Add tests for kernel_bundle API

### PR DESCRIPTION
**Pull request by aleksmesh**
Add tests according to [2.2. sycl::kernel_bundle API](https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/test_plans/kernel_bundle.asciidoc#22-syclkernel_bundle-api) of test plan.

The tests for kernel_bundle::has_kernel and kernel_bundle::get_kernel are disabled until implemented.

### CODE REVIEW

**Review by ProGTX — APPROVED** [review]

**Comment by fraggamuffin** [comment]
LGTM

**Review by keryell — APPROVED** [review]
Thanks.

File: tests/kernel_bundle/CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,7 @@ set(independent_cases_list
   get_kernel_bundle_zero_kernels.cpp
   get_kernel_bundle_with_kernel_attr.cpp
   get_kernel_bundle_without_kernel_attr.cpp
+  kernel_bundle_api.cpp
```
Comment (bader) [inline tests/kernel_bundle/CMakeLists.txt:29]: Just for my own education: why `kernel_bundle_api` is this list, but `kernel_bundle_spec_const` from #493 is not?

Comment (aleksmesh) [reply tests/kernel_bundle/CMakeLists.txt]: Including kernel_bundle_api in the list is my mistake. Fixed it, thanks.

**Pull request merged**